moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -136,21 +138,26 @@ void planning_scene_monitor::CurrentStateMonitor::clearUpdateCallbacks()
 
 void planning_scene_monitor::CurrentStateMonitor::startStateMonitor(const std::string& joint_states_topic)
 {
+  rclcpp::Clock ros_clock;
   if (!state_monitor_started_ && robot_model_)
   {
     joint_time_.clear();
-    if (joint_states_topic.empty())
-      ROS_ERROR("The joint states topic cannot be an empty string");
+    if (joint_states_topic.empty()){
+      RCLCPP_ERROR(logger, " The joint states topic cannot be an empty string");
+    }
     else
-      joint_state_subscriber_ = nh_.subscribe(joint_states_topic, 25, &CurrentStateMonitor::jointStateCallback, this);
+      auto joint_state_subscriber_ = node_->create_subscription<sensor_msgs::msg::JointState>(joint_states_topic,
+                        std::bind(&planning_scene_monitor::CurrentStateMonitor::jointStateCallback, this, std::placeholders::_1));
     if (tf_buffer_ && !robot_model_->getMultiDOFJointModels().empty())
     {
       tf_connection_.reset(new TFConnection(
-          tf_buffer_->_addTransformsChangedListener(boost::bind(&CurrentStateMonitor::tfCallback, this))));
```

Comment:
@davetcoleman and  @mlautman I'm trying to replace the _addTransformsChangedListener call since on ros2 geometry this function is not implemented, I would like some input about how to procede with that. 

There is a similar function called [addTransformableCallback](http://docs.ros.org/jade/api/tf2/html/classtf2_1_1BufferCore.html#a71a56dfdd102295b339684406218f986) but it has nothing to do with that.

Reply:
ping @henningkayser, @davetcoleman, @mlautman

Reply:
Do you know why ``_addTransformsChangedListener()`` is not implemented?

@IanTheEngineer made this change 10 months ago, here: https://github.com/ros-planning/moveit/commit/736251dc77a24877135522eb901ae40a1ea9872b

Looking at that [PR](https://github.com/ros-planning/moveit/pull/830), it required the following PRs to pass:

This PR depends on the following upstream API changes for Melodic:
https://github.com/ros/geometry/pull/163 (Exposing `tf2_ros::Buffer` object from TransformListener)
https://github.com/ros/geometry2/pull/292 (tf2 Type conversions)
https://github.com/ros/geometry2/pull/294 (Additional tf2 Type conversions)
https://github.com/ros-visualization/rviz/pull/1236 (Exposing tf2 without ROS deprecation warnings)

So likely one of those needs to be forward ported to ROS2.

Reply:
> Do you know why `_addTransformsChangedListener()` is not implemented?

Look at the https://github.com/ros-planning/moveit/pull/830#issuecomment-389881823, on the porting procedure they just left it apart as it seems to be "deprecated".

Reply:
https://github.com/ros-planning/moveit/pull/830#issuecomment-386517416
> addTransformsChangedListener API has been replaced by the TransformableRequest interface. An example of using it is in the tf2::MessageFilters: http://docs.ros.org/jade/api/tf2_ros/html/c++/message__filter_8h_source.html

Reply:
@IanTheEngineer can you help shed some light on this question?

Reply:
ping @mlautman, @davetcoleman 

I have to finish this...

Reply:
@anasarrak I don't know if I have any more information than you do at this point.. Have you had the chance to hunt down the dependencies that Dave referenced to see if any of them might be holding up the port? Perhaps @tfoote's comment can help shed some light here: https://github.com/ros-planning/moveit/pull/830#issuecomment-386517416